cfme/tests/services/test_dialog_regex_validation_in_catalog.py

```diff
@@ -55,20 +55,59 @@ def dialog_cat_item(appliance, catalog, request):
     sd.delete_if_exists()
 
 
-@pytest.mark.parametrize("dialog_cat_item", [{"validation": "^([a-z0-9]+_*)*[a-z0-9]+$"}],
-                         indirect=True)
-def test_dialog_element_regex_validation(appliance, dialog_cat_item):
-    """Tests Service Dialog Elements with regex validation.
+@pytest.fixture(scope="function")
+def dialog_textarea_cat_item(appliance, catalog, request):
 
-    Testing BZ 1518971
+    service_dialog = appliance.collections.service_dialogs
+    dialog = fauxfactory.gen_alphanumeric(12, start="dialog_")
+
+    element_data = {
+        'element_information': {
+            'ele_label': fauxfactory.gen_alphanumeric(15, start="ele_label_"),
+            'ele_name': fauxfactory.gen_alphanumeric(15, start="ele_name_"),
+            'ele_desc': fauxfactory.gen_alphanumeric(15, start="ele_desc_"),
+            'choose_type': "Text Area"
+        },
+        'options': {
+            'validation_switch': True,
+            'validation': request.param["validation"]
+        }
+    }
+
+    if appliance.version < '5.10':
+        element_data["options"].pop("validation_switch", None)
+    sd = service_dialog.create(label=dialog, description="my dialog")
+    tab = sd.tabs.create(tab_label=fauxfactory.gen_alphanumeric(start="tab_"),
+                         tab_desc="my tab desc")
+    box = tab.boxes.create(box_label=fauxfactory.gen_alphanumeric(start="box_"),
+                           box_desc="my box desc")
+    box.elements.create(element_data=[element_data])
+
+    catalog_item = appliance.collections.catalog_items.create(
+        appliance.collections.catalog_items.GENERIC,
+        name=fauxfactory.gen_alphanumeric(15, start="cat_item_"),
+        description="my catalog",
+        display_in=True,
+        catalog=catalog,
+        dialog=sd)
+    yield catalog_item, element_data, sd
+    if catalog_item.exists:
+        catalog_item.delete()
+    sd.delete_if_exists()
+
+
+@pytest.mark.parametrize("dialog_textarea_cat_item", [{"validation": "^([a-z0-9]+_*)*[a-z0-9]+$"}],
+                         indirect=True)
+def test_dialog_text_area_regex_validation(appliance, dialog_textarea_cat_item):
```

Comment:
From what I can tell, the only difference between this test and `test_dialog_element_regex_validation` are the fixture parameters. Is there any way we can combine these test cases?

Reply:
I will give a try